tests/user/test_user_request_builder.py

```diff
@@ -140,6 +140,30 @@ def test_user_admin_build_add_request_with_bad_segment_traits(self):
             + f"combination for '{self.user_admin._profile_type}'.\n",
         )
 
+    def test_user_admin_cleans_up_after_build_add_request_with_bad_segment_traits(self):
+        bad_trait = "omvs:bad_trait"
+        user_admin = UserAdmin(
+            generate_requests_only=True,
+        )
+        with self.assertRaises(SegmentTraitError) as exception:
+            user_admin.add(
+                "squidwrd", TestUserConstants.TEST_ADD_USER_REQUEST_BAD_TRAITS
+            )
+        self.assertEqual(
+            exception.exception.message,
+            "Unable to build Security Request.\n\n"
+            + f"'{bad_trait}' is not a known segment-trait "
+            + f"combination for '{self.user_admin._profile_type}'.\n",
+        )
+        result = user_admin.extract("squidwrd", segments=["omvs"])
```

Comment:
Is there a reason for doing extract and add here?

Reply:
Yes, in GenerateRequestOnly Mode, Add does not do a preliminary extract, but outside of this mode it does to know if it needs to throw AddOperationError(s). This extract emulates that, as this is necessary to see the impact of the bug.

Reply:
Can you add a comment?